fpmsyncd/routesync.cpp

```diff
@@ -756,88 +790,143 @@ void RouteSync::onRouteMsg(int nlmsg_type, struct nl_object *obj, char *vrf)
+            FieldValueTuple gw("nexthop", nexthops.c_str());
+            FieldValueTuple intf("ifname", ifnames.c_str());
+            fvVector.push_back(gw);
+            fvVector.push_back(intf);
+            SWSS_LOG_DEBUG("NextHop group id %d is a single nexthop address. Filling the route table %s with nexthop and ifname", nhg_id, destipprefix);
+        }
+        else
+        {
+            nhg_id_key = getNextHopGroupKeyAsString(nhg_id);
+            FieldValueTuple nhg("nexthop_group", nhg_id_key.c_str());
+            fvVector.push_back(nhg);
+            updateNextHopGroup(nhg_id);
+        }
+
+        auto proto_num = rtnl_route_get_protocol(route_obj);
+        auto proto_str = getProtocolString(proto_num);
+        FieldValueTuple proto("protocol", proto_str);
+        fvVector.push_back(proto);
+
+    }
+    else
+#endif
+    {
+        struct nl_list_head *nhs = rtnl_route_get_nexthops(route_obj);
+        if (!nhs)
+        {
+            SWSS_LOG_INFO("Nexthop list is empty for %s", destipprefix);
+            return;
+        }
+
+        /* Get nexthop lists */
+
+        getNextHopList(route_obj, gw_list, mpls_list, intf_list);
+        string weights = getNextHopWt(route_obj);
+
+        vector<string> alsv = tokenize(intf_list, NHG_DELIMITER);
+        for (auto alias : alsv)
         {
-            SWSS_LOG_DEBUG("Skip routes to eth0 or docker0: %s %s %s",
-                    destipprefix, gw_list.c_str(), intf_list.c_str());
-            // If intf_list has only this interface, that means all of the next hops of this route 
-            // have been removed and the next hop on the eth0/docker0 has become the only next hop. 
-            // In this case since we do not want the route with next hop on eth0/docker0, we return. 
-            // But still we need to clear the route from the APPL_DB. Otherwise the APPL_DB and data 
-            // path will be left with stale route entry
-            if(alsv.size() == 1)
+            /*
+             * An FRR behavior change from 7.2 to 7.5 makes FRR update default route to eth0 in interface
+             * up/down events. Skipping routes to eth0 or docker0 to avoid such behavior
+             */
+            if (alias == "eth0" || alias == "docker0")
             {
-                if (!warmRestartInProgress)
+                SWSS_LOG_DEBUG("Skip routes to eth0 or docker0: %s %s %s",
+                        destipprefix, gw_list.c_str(), intf_list.c_str());
+                // If intf_list has only this interface, that means all of the next hops of this route
+                // have been removed and the next hop on the eth0/docker0 has become the only next hop.
+                // In this case since we do not want the route with next hop on eth0/docker0, we return.
+                // But still we need to clear the route from the APPL_DB. Otherwise the APPL_DB and data
+                // path will be left with stale route entry
+                if(alsv.size() == 1)
```

Comment:
Can't this test be moved outside the loop ? If the list is single entry then there is no reason for the loop itself.

Reply:
The purpose of this loop is not to show skipped routes, but to skip routes to specific interfaces (eth0 or docker0) and do the associated processing.